### PR TITLE
ETH2 bridger refactor

### PR DESCRIPTION
**Pull request by furoxr**
### ETH2 bridger refactor
1. Common traits are in bridge-e2e crate
2. Bridger logic is in relay-e2e crate
3. Fee market and simple fee market strategy is in feemarket-e2e crate
4. Contract interfaces are in client-contracts crate

### CODE REVIEW

**Review by HackFisher — COMMENTED** [review]
@hujw77 Can we start test on test net?

**Comment by HackFisher** [comment]
@furoxr please resolve the conflicts, cc @fewensa

**Comment by fewensa** [comment]
https://github.com/darwinia-network/bridger/issues/524
@furoxr please also change the config name 

https://github.com/darwinia-network/bridger/blob/ac10bfe82e43445e192311a27f2bb63e6862aa08/bridges/pangoro-goerli/bridge/src/bridge/config.rs#L32-L33

File: assistants/relay-e2e/src/header/eth_beacon_header_relay.rs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+            tokio::time::sleep(std::time::Duration::from_secs(10)).await;
+        }
+    }
+
+    pub async fn run(&mut self) -> RelayResult<()> {
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map_err(|e| RelayError::Custom(format!("{}", e)))?
+            .as_secs();
+        if now - self.last_relay_time <= self.minimal_interval {
```
Comment (fewensa) [inline assistants/relay-e2e/src/header/eth_beacon_header_relay.rs:50]: A simple way is use `config.general.header_relay_minimum_interval` in tokio::time::sleep(std::time::Duration::from_secs(10)).await;

tokio::time::sleep(std::time::Duration::from_secs(config.general.header_relay_minimum_interval)).await;

you don't need to write this if condition.

**Review by fewensa — APPROVED** [review]

**Review by HackFisher — APPROVED** [review]
@furoxr We are still waiting another PR for fixing:
1. Support and adaption of nimbus nodes apis
2. Check recent loadstar changes https://github.com/ethereum/consensus-specs/pull/2932

**Pull request merged**